kolibri/plugins/coach/assets/src/state/actions/exam.js

```diff
@@ -104,9 +146,131 @@ function showExamsPage(store, classId) {
+  ExamResource.getModel(examId).save({ active: true }).then(
+    () => {
+      const exams = store.state.pageState.exams;
+      const examIndex = exams.findIndex(exam => exam.id === examId);
+      exams[examIndex].active = true;
+
+      store.dispatch('SET_EXAMS', exams);
+      this.displayModal(false);
+    },
+    error => CoreActions.handleError(store, error)
+  );
+}
+
+function deactivateExam(store, examId) {
+  ExamResource.getModel(examId).save({ active: false }).then(
+    () => {
+      const exams = store.state.pageState.exams;
+      const examIndex = exams.findIndex(exam => exam.id === examId);
+      exams[examIndex].active = false;
+
+      store.dispatch('SET_EXAMS', exams);
+      this.displayModal(false);
+    },
+    error => CoreActions.handleError(store, error)
+  );
+}
+
+function _assignExamTo(examId, collection) {
+  const assignmentPayload = {
+    exam: examId,
+    collection,
+  };
+  return new Promise((resolve, reject) => {
+    ExamAssignmentResource.createModel(assignmentPayload).save().then(
+      assignment => resolve(assignment),
+      error => reject(error)
+    );
+  });
+}
+
+function _removeAssignment(assignmentId) {
+  return new Promise((resolve, reject) => {
+    ExamAssignmentResource.getModel(assignmentId).delete().then(
+      () => resolve(),
+      error => reject(error)
+    );
+  });
+}
+
+function updateExamAssignments(store, examId, collectionsToAssign, assignmentsToRemove) {
+  const assignPromises = collectionsToAssign.map(collection => _assignExamTo(examId, collection));
+  const unassignPromises = assignmentsToRemove.map(assignment => _removeAssignment(assignment));
+  const assignmentPromises = assignPromises.concat(unassignPromises);
+
+  ConditionalPromise.all(assignmentPromises).only(CoreActions.samePageCheckGenerator(store),
+    response => {
+      let newAssignments = response.filter(n => n);
+      newAssignments = _assignmentsState(newAssignments);
+
+      const classId = store.state.pageState.currentClass.id;
+      const exams = store.state.pageState.exams;
+      const examIndex = exams.findIndex(exam => exam.id === examId);
+      const examVisibility = exams[examIndex].visibility;
+
+      newAssignments.forEach(assignment => {
+        if (assignment.collection.id === classId) {
+          examVisibility.class = assignment;
+        } else {
+          examVisibility.groups.push(assignment);
+        }
+      });
+
+      assignmentsToRemove.forEach(assignmentId => {
+        if (examVisibility.class) {
+          if (assignmentId === examVisibility.class.assignmentId) {
+            examVisibility.class = null;
+            return;
+          }
+        }
+        examVisibility.groups = examVisibility.groups.filter(
+          group => group.assignmentId !== assignmentId);
+      });
+
+      exams[examIndex].visibility = examVisibility;
+      store.dispatch('SET_EXAMS', exams);
+      store.dispatch('CORE_SET_ERROR', null);
+      this.displayModal(false);
```

Comment:
This method does not exist here (looks like it has been copied from the code for a component).